specs/AdaptiveCardsXamarinSpec.md

```diff
@@ -0,0 +1,188 @@
+
+#### ApiDefinitions
+``` C#
+// @interface ACOBaseActionElement : NSObject
+[BaseType(typeof(NSObject))]
+interface ACOBaseActionElement
+{
+    // @property ACRActionType type;
+    [Export("type", ArgumentSemantic.Assign)]
+    ACRActionType Type { get; set; }
+
+    // -(NSString *)title;
+    [Export("title")]
+    string Title { get; }
+
+    // -(NSString *)elementId;
+    [Export("elementId")]
+    string ElementId { get; }
+
+    // -(NSString *)url;
+    [Export("url")]
+    string Url { get; }
+
+    // -(NSString *)data;
+    [Export("data")]
+    string Data { get; }
+
+    // -(NSData *)additionalProperty;
+    [Export("additionalProperty")]
+    NSData AdditionalProperty { get; }
+}
+```
+
+#### Structs
+``` C#
+[Native]
+public enum ACRActionType : long
+{
+	ShowCard = 1,
+	Submit,
+	OpenUrl
+}
+```
+
+### APIs to be included in Xamarin
+
+| API Name | iOS API | Priority |
+| --- | --- | --- |
+| Parsing | ```(ACOAdaptiveCardParseResult*) fromJson: (NSString*)``` | P0 |
+| Rendering | ```(ACRRenderResult*) render:(ACOAdaptiveCard*) config:(ACOHostConfig) widthConstraint:(float)``` | P0 |
+| Rendering | ```(ACRRenderResult*) render:(ACOAdaptiveCard*) config:(ACOHostConfig) widthConstraint:(float) delegate:(id<ACRActionDelegate>)``` | P0 |
+| Rendering | ```(ACRRenderResult*) renderAsViewController:(ACOAdaptiveCard*) config:(ACOHostConfig) frame:(CGRect) delegate:(id<ACRActionDelegate>)``` | P0 |
+| Custom Element | ```(void) setBaseCardElementRenderer: (ACRBaseCardElementRenderer*)cardElementType: (ACRCardElementType)``` | P1 |
+| Custom Element | ```(void) setCustomElementParser: (NSObject<ACOIBaseCardElementParser>)``` | P1 |
+| Custom Element | ```(void) setCustomElementParser: (NSObject<ACOIBaseCardElementParser>) key: (NSString*)``` | P1 |
+| Custom Element | ```(void) setCustomElementRenderer: (ACRBaseCardElementRenderer*) key:(NSString *)``` | P1 |
+| Custom Element | ```(NSObject<ACOIBaseCardElementParser>*) getCustomElementParser: (NSString*)``` | P1 |
+| Custom Element | ```(ACRBaseCardElementRenderer*) getRenderer: (NSNumber *)``` | P1 |
+| Custom Action | ```(void) setActionRenderer: (ACRBaseActionElementRenderer) cardElementType: (NSNumber*)``` | P1 |
+| Custom Action | ```(ACRBaseActionElementRenderer*) getActionRenderer: (NSNumber*)``` | P1 |
+| Custom Action Set | ```(void) setActionSetRenderer: (id<ACRIBaseActionSetRenderer>)``` | P2 |
+| Custom Action Set | ```(id<ACRIBaseActionSetRenderer>) getActionSetRenderer``` | P2 | 
+| Resource Resolvers | ```(void) setResourceResolver: (NSObject<ACOIResourceResolver>) scheme: (NSString*)``` | P2 |
+| Resource Resolvers | ```(NSObject<ACOIResourceResolver>) getResourceResolverForScheme: (NSString*)``` | P2 |
+| Feature Registration | ```(void) addFeature: (NSString*) featureVersion: (NSString*)``` | P0 | 
+| Feature Registration | ```(NSString*) removeFeature: (NSString*)``` | P0 |
```

Comment:
[Team Spec Review] Can you file a workitem to add documentation/blog content around Feature Registration vs Resource Resolver use cases? @Paulcam206